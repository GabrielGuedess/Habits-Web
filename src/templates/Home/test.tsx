import { render, screen } from '@testing-library/react';

import { Home } from '.';

jest.mock('components/molecules/Header', () => ({
  __esModule: true,
  Header: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Header Mock">{children}</div>;
  },
}));

jest.mock('components/organisms/SummaryTable', () => ({
  __esModule: true,
  SummaryTable: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="SummaryTable Mock">{children}</div>;
  },
}));

describe('<Home />', () => {
  it('should render the Home', () => {
    const { container } = render(<Home />);

    expect(screen.getByTestId('Header Mock')).toBeInTheDocument();
    expect(screen.getByTestId('SummaryTable Mock')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
