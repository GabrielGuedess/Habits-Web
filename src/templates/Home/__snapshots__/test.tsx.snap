// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the Home 1`] = `
<div
  class="flex h-screen w-screen items-center justify-center"
>
  <div
    class="flex w-full max-w-5xl flex-col gap-16 px-6"
  >
    <div
      data-testid="Header Mock"
    />
    <div
      data-testid="SummaryTable Mock"
    />
  </div>
</div>
`;
